### PR TITLE
tests: Skips Windows-unrelated tests on Windows

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig windows
/sig testing

**What this PR does / why we need it**:

Skips IPv6 tests on Windows.
Skips sysctl tests on Windows.
Skips network policy tests on Windows.
Skips RunAsUser / FSGroup / file permissions related tests, as those are not supported on Windows.
Skips the test "should preserve source pod IP for traffic thru service cluster IP" on Windows, as it creates a Pod with ``HostNetwork=true``, which is unsupported.

What works and what doesn't work on Windows has been documented here:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#windows--linux-considerations

**Which issue(s) this PR fixes**:

Related: #69871

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/framework/util.go
Comment (neolit123) [inline test/e2e/framework/util.go]: i don't think the variadic string list is needed here, but looks like this is following the existing pattern.

**Comment by neolit123** [comment]
i've reopened https://github.com/kubernetes/kubernetes/issues/69871 as it matches the issue name.
please close it if this PR finishes the work.

/priority important-longterm

**Comment by neolit123** [comment]
/approve
on the framework change and LGTM overall.

please assign approvers for the rest or find a top level approver.

**Review by spiffxp — CHANGES_REQUESTED** [review]
Thanks, really appreciate the explanatory comments you dropped

Since you've gone to the trouble of identifying these, can you please tag these tests as `[LinuxOnly]` per [Kinds of Tests](https://github.com/kubernetes/community/blob/40c7cc99110a2634ab52d0e540eeb5d483bce2e6/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests)?

**Comment by spiffxp** [comment]
needs rebase

**Comment by bclau** [comment]
> 
> 
> Thanks, really appreciate the explanatory comments you dropped
> 
> Since you've gone to the trouble of identifying these, can you please tag these tests as `[LinuxOnly]` per [Kinds of Tests](https://github.com/kubernetes/community/blob/40c7cc99110a2634ab52d0e540eeb5d483bce2e6/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests)?

Done

**Comment by spiffxp** [comment]
/approve
will need a rebase before I can lgtm

**Comment by johnbelamaric** [comment]
Seems like we might want to add something in the framework (wrapper on It?) that will automatically skip tests marked [LinuxOnly] in windows environments - and the same thing could automatically skip based on the [Provider:X] tags.

**Comment by bclau** [comment]
> 
> 
> Seems like we might want to add something in the framework (wrapper on It?) that will automatically skip tests marked [LinuxOnly] in windows environments - and the same thing could automatically skip based on the [Provider:X] tags.

Well, I've typically added the ``[LinuxOnly]`` tag to Conformance tests, since there is a rule that we cannot have any skip logic in those tests, so those tests are excluded through regex. Adding that wrapper will only be useful until the tests are promoted to Conformance.

Outside of Conformance tests, this might be the first time I've added the ``[LinuxOnly]`` tag to tests.

We could still add that wrapper, but its usecase is quite limited IMO.

**Pull request merged**
